code/modules/xenoarchaeology/tools/xenoarchaeology_machines.dm

```diff
@@ -0,0 +1,323 @@
+/*
+	Misc machines used to interact with artifact traits
+*/
+
+/obj/machinery/xenoarchaeology_machine
+	icon = 'icons/obj/xenoarchaeology/xenoartifact_tech.dmi'
+	///Do we move the artifact to our turf, or inside us?
+	var/move_inside = FALSE
+	///List of things we need to spit out
+	var/list/held_contents = list()
+	var/max_contents = 1
+
+/obj/machinery/xenoarchaeology_machine/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_ARTIFACT_IGNORE, GENERIC_ITEM_TRAIT)
+
+/obj/machinery/xenoarchaeology_machine/attackby(obj/item/I, mob/living/user, params)
```

Comment:
this is actually okay for using arbitrary name `I` here, because it's from the proc parameter. just be aware of that.